src/stub/src/powerpc64le-linux.elf-main.c

```diff
@@ -331,7 +339,9 @@ void *upx_main(
   { // Map PT_INTERP program interpreter
     int j;
     for (j=0; j < ehdr->e_phnum; ++phdr, ++j) if (PT_INTERP==phdr->p_type) {
-        char const *const iname = (char const *)phdr->p_vaddr;
+        char *iname = (char *)phdr->p_vaddr;
+        if ( phdr->p_vaddr <= (entry & PAGE_MASK )) // pie relocatable addr ?
```

Comment:
This logic is not reliable.  There is no necessary ordering between the entry point and where the name of the interpreter is stored.  In particular, if the -fpie (ET_DYN) executable has been pre-linked (slid to a non-zero base address before execution) then the ordering between .p_vaddr and entry can change.

Please see stub/src/amd64-linux.elf-main.c:
```
    entry = do_xmap(ehdr, &xi1, 0, av, f_decompress, f_unf, &reloc);  // "rewind"
```
where the new last parameter communicates the "slide" value between do_xmap() and upx_main().  Similar code should be used.